storybook/stories/API/cartesian/ReferenceLine.stories.tsx

```diff
@@ -0,0 +1,199 @@
+    table: { type: { summary: 'string | number' }, category: 'General' },
+  },
+  x: {
+    description: `If set a string or a number, a vertical line perpendicular to the x-axis specified by xAxisId 
+    will be drawn. If the specified x-axis is a number axis, the type of x must be Number. If the specified x-axis 
+    is a category axis, the value of x must be one of the categories, otherwise no line will be drawn.`,
+    table: { type: { summary: 'number | string' }, category: 'General' },
+  },
+  y: {
+    description: `If set a string or a number, a horizontal line perpendicular to the y-axis specified by yAxisId 
+    will be drawn. If the specified y-axis is a number axis, the type of y must be Number. If the specified y-axis 
+    is a category axis, the value of y must be one of the catagories, otherwise no line will be drawn.`,
+    table: { type: { summary: 'number | string' }, category: 'General' },
+  },
+  xAxis: {
+    description: 'The configuration of the corresponding x-axis, usually calculated internally.',
+    table: { type: { summary: 'object' }, category: 'General' },
+  },
+  yAxis: {
+    description: 'The configuration of the corresponding y-axis, usually calculated internally.',
+    table: { type: { summary: 'object' }, category: 'General' },
+  },
+  ifOverflow: {
+    description: `Defines how to draw the reference line if it falls partly outside the canvas. If set to 'discard', 
+    the reference line will not be drawn at all. If set to 'hidden', the reference line will be clipped to the canvas. 
+    If set to 'visible', the reference line will be drawn completely. If set to 'extendDomain', the domain of the 
+    overflown axis will be extended such that the reference line fits into the canvas.`,
+    table: { type: { summary: "'discard' | 'hidden' | 'visible' | 'extendDomain'" }, category: 'General' },
+    default: 'discard',
+  },
+};
+
+const LabelProps: Args = {
+  label: {
+    description: `If set a string or a number, default label will be drawn, and the option is content. 
+    If set a React element, the option is the custom react element of drawing label. If set a function, 
+    the function will be called to render customized label.`,
+    table: { type: { summary: 'string | number | ReactElement | Function' }, category: 'Label' },
+  },
+};
+
+const StyleProps: Args = {
+  stroke: {
+    control: { type: 'color' },
+    table: { category: 'Style' },
+    defaultValue: '#ccc',
+  },
+  strokeWidth: {
+    description: 'The width of the stroke.',
+    table: {
+      type: {
+        summary: 'string | number',
+      },
+      category: 'Style',
+    },
+    defaultValue: 1,
+  },
+  strokeDasharray: {
+    description: `The pattern of dashes and gaps used to paint the line.
+      https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-dasharray`,
+    table: {
+      type: {
+        summary: 'string',
+      },
+      category: 'Style',
+    },
+  },
```

Comment:
Should we possibly reuse the existing documentation instead of redefining it here? i.e. maybe export the style props separately from `props/Style.ts`